### ISSUE TITLE
`management.tracing.enabled=false` does not disable tracing propagation in Spring Boot 3.2.0

### ISSUE THREAD

**Issue by lkleeven** [spring-projects/spring-boot#38625]
Before Spring Boot 3.2.0 setting the `management.tracing.enabled` property to false would also disable the underlying tracing propagation. However with Spring Boot 3.2.0 things have changed with regards to observability. As far as I understand this should have only affected how to test observability.

However, if I now run an application with tracing support and set the `management.tracing.enabled` to false I can see that the application still propagates tracing headers (or starts a new trace). I expected that setting `management.tracing.enabled` to false would also disable tracing propagation (and starting new traces).

I've reproduced the problem in [this minimal Spring Boot project](https://github.com/lkleeven/spring-boot-tracing-propagation). The project has an included unit test that now fails. You can run it with `.mvnw clean verify`. Alternatively, you can run the application and do a call to the `/demo` endpoint which should return `true` but now returns `false` due to the suspected bug.

I've based my reproduction on a simple premise where you can call the `/demo` endpoint, which will then call another endpoint. That one checks if the `traceparent` header has been set. If so it returns `false` because we expect it not to be set, otherwise it returns `true`. In the applications properties the `management.tracing.enabled` property is set to `false`. Extra logging is present to help in debugging.

**Comment by mhalbritter** [comment]
`management.tracing.enabled` was never intended to switch off tracing. Its sole purpose was to disable tracing in tests. In hindsight it was a mistake that it ended up in the documentation.

As we restructured the bean arrangement in https://github.com/spring-projects/spring-boot/issues/35354, now only the reporting beans are disabled when `management.tracing.enabled` is false. The beans which do the propagation are still in place.

If you want to disable propagation or tracing, please remove the dependencies from the classpath. Generally in Boot, we don't have `enabled` properties if there's a better signal, which in this case is the presence of classes on the classpath.

**Comment by lkleeven** [comment]
In that case, could you at least update the documentation for the property?

**Comment by mhalbritter** [comment]
Sure, i've opened https://github.com/spring-projects/spring-boot/issues/38626 for that.

**Comment by lkleeven** [comment]
Too bad that there won't be a property to disable this. In our experience providing a Spring Boot based framework to our company, having such properties has been invaluable to quickly help teams progress. If only to allow the option to quickly disable functionality in case of troubleshooting or production issues.

**Comment by lkleeven** [comment]
They might not all have the ability to disable whole feature sets, but the [properties documentation](https://docs.spring.io/spring-boot/docs/3.2.0/reference/html//application-properties.html) shows 152 `enabled` properties. Which we are happy users of because, like I said, this makes it much easier to test and troubleshoot issues.

**Comment by mhalbritter** [comment]
You're right. These properties enable/disable features which are not detectable via other means.

**Comment by quaff** [comment]
> You won't find a lot of `enabled` properties in Spring Boot. Features are usually enabled if a certain JAR is on the classpath. If you don't want a particular feature, either remove the JAR from the classpath or [disable the auto-configuration](https://docs.spring.io/spring-boot/docs/current/reference/htmlsingle/#using.auto-configuration.disabling-specific).

Then we need to build different artifacts for different deployments, I'd like build once and run for all by disable/enable feature flag.

**Comment by mhalbritter** [comment]
You can also disable auto-configuration with a property, [see here](https://docs.spring.io/spring-boot/docs/current/reference/htmlsingle/#using.auto-configuration.disabling-specific).

**Comment by mhalbritter** [comment]
All auto-configurations work like that, I don't see why the auto-configuration for tracing should have a feature flag while the others don't.

**Comment by quaff** [comment]
> All auto-configurations work like that, I don't see why the auto-configuration for tracing should have a feature flag while the others don't.

I'm discussing enable/disable feature by including dependency. But it's more desired for tracing, It's very common that an application need tracing for production deployment but not needed for SIT/UAT deployment.

**Comment by mhalbritter** [comment]
Hey @lkleeven, we talked about that tracing disabling in our meeting and decided to make that `management.tracing.enabled` flag more official. What it does in 3.2.1 is that it disables span exporting (sending traces to your backend) and also trace propagation over the wire (e.g. HTTP). See https://github.com/spring-projects/spring-boot/issues/38641 for details.
